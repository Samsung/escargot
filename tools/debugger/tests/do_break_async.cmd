s
bt
b do_break_async.js:40
b do_break_async.js:31
b do_break_async.js:22
c
bt
c
bt
c
bt
c
