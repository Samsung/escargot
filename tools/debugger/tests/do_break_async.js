/*
 * Copyright (c) 2021-present Samsung Electronics Co., Ltd
 *
 *  This library is free software; you can redistribute it and/or
 *  modify it under the terms of the GNU Lesser General Public
 *  License as published by the Free Software Foundation; either
 *  version 2.1 of the License, or (at your option) any later version.
 *
 *  This library is distributed in the hope that it will be useful,
 *  but WITHOUT ANY WARRANTY; without even the implied warranty of
 *  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
 *  Lesser General Public License for more details.
 *
 *  You should have received a copy of the GNU Lesser General Public
 *  License along with this library; if not, write to the Free Software
 *  Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301
 *  USA
 */

async function f() {
    await 0
    a = 0
}

function g() {
    f(0)
}

async function h() {
    await 0
    g()
}

function i() {
    h(0)
}

async function j() {
    await 0
    i()
}

j()
