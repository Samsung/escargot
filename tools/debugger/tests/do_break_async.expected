Connecting to: localhost:6501
Connection created!!!
Stopped at tools/debugger/tests/do_break_async.js:43
(escargot-debugger) s
Stopped at tools/debugger/tests/do_break_async.js:39 (in j() at line:38, col:1)
(escargot-debugger) bt
tools/debugger/tests/do_break_async.js:39:5 [depth:0] (in j)
tools/debugger/tests/do_break_async.js:43:1 [depth:1]
(escargot-debugger) b do_break_async.js:40
Breakpoint 1 at tools/debugger/tests/do_break_async.js:40 (in j() at line:38, col:1)
(escargot-debugger) b do_break_async.js:31
Breakpoint 2 at tools/debugger/tests/do_break_async.js:31 (in h() at line:29, col:1)
(escargot-debugger) b do_break_async.js:22
Breakpoint 3 at tools/debugger/tests/do_break_async.js:22 (in f() at line:20, col:1)
(escargot-debugger) c
Stopped at breakpoint:1 tools/debugger/tests/do_break_async.js:40 (in j() at line:38, col:1)
(escargot-debugger) bt
tools/debugger/tests/do_break_async.js:40:5 [depth:0] (in j)
tools/debugger/tests/do_break_async.js:43:1
(escargot-debugger) c
Stopped at breakpoint:2 tools/debugger/tests/do_break_async.js:31 (in h() at line:29, col:1)
(escargot-debugger) bt
tools/debugger/tests/do_break_async.js:31:5 [depth:0] (in h)
tools/debugger/tests/do_break_async.js:35:5 (in i)
tools/debugger/tests/do_break_async.js:40:5 (in j)
tools/debugger/tests/do_break_async.js:43:1
(escargot-debugger) c
Stopped at breakpoint:3 tools/debugger/tests/do_break_async.js:22 (in f() at line:20, col:1)
(escargot-debugger) bt
tools/debugger/tests/do_break_async.js:22:5 [depth:0] (in f)
tools/debugger/tests/do_break_async.js:26:5 (in g)
tools/debugger/tests/do_break_async.js:31:5 (in h)
tools/debugger/tests/do_break_async.js:35:5 (in i)
tools/debugger/tests/do_break_async.js:40:5 (in j)
tools/debugger/tests/do_break_async.js:43:1
(escargot-debugger) c
Connection closed.
